platform/darwin/src/MGLShapeCollection.mm

```diff
@@ -13,6 +14,18 @@ + (instancetype)shapeCollectionWithShapes:(NSArray<MGLShape *> *)shapes {
 - (instancetype)initWithShapes:(NSArray<MGLShape *> *)shapes {
     if (self = [super init]) {
         _shapes = shapes.copy;
+        
+        for (MGLShape* shape in shapes ){
+            if ([shape conformsToProtocol:@protocol(MGLFeature)]) {
```

Comment:
Putting the feature check here does address a common mistake, but #11287 was specifically about warning when creating an `MGLShapeSource` with an MGLShapeCollection or setting `MGLShapeSource.shape` to an MGLShapeCollection. That would occur later than this code; the only difference is that there may theoretically be a legitimate reason to stuff MGLFeatures in an MGLShapeCollection, but there’s no reason to stick an MGLShapeCollection inside an MGLShapeSource or MGLComputedShapeSource. On the other hand, it’s always nicer to warn as soon as we see a potential problem unfolding. Your call as to the proper tradeoff.